src/midi/midi_message_publisher.rs

```diff
@@ -0,0 +1,33 @@
+use crate::Message;
+use std::sync::mpsc::{self, Receiver, Sender};
+use std::thread::{self, JoinHandle};
+
+pub struct MidiMessagePublisher {
+    midi_messages: Receiver<Message>,
+    senders: Vec<Sender<Message>>,
```

Comment:
Where I landed with `MidiMessagePublisher` (which technically isn't even necessary yet because there's only one consumer of the incoming MIDI "input" messages (the duration slider listener), but the thought was that it shouldn't own that MIDI message receiver, there should be a pattern for multiple consumers of the same MIDI message stream) was to make it just "multiply" incoming MIDI messages onto multiple output channels (there must be a name for this pattern)

Reply:
The standard library doesn't include an MPMC channel for some reason, but the popular crossbeam library does: https://docs.rs/crossbeam/0.8.1/crossbeam/channel/index.html. (I've seen this called a "broadcast" channel elsewhere, also.)

Reply:
Ok looking at the crossbeam docs those multiple listeners seem to "compete" for the single message stream, whereas what I want here is multiple copies of the message stream (so each consumer gets a "full stream")

So I saw [`bus`](https://docs.rs/bus/2.2.3/bus/) mentioned and updated to use it